tests/extensions/test_render.py

```diff
@@ -0,0 +1,234 @@
+        rescale=[[0, 1000], [0, 1000], [0, 1000]],
+        nodata=-9999,
+        colormap_name="viridis",
+        colormap={
+            "0": "#e5f5f9",
+            "10": "#99d8c9",
+            "255": "#2ca25f",
+        },
+        color_formula="gamma rg 1.3, sigmoidal rgb 22 0.1, saturation 1.5",
+        resampling="bilinear",
+        expression="(B08-B04)/(B08+B04)",
+        minmax_zoom=[2, 18],
+    )
+
+
+@pytest.fixture
+def thumbnail_render(ext_item: pystac.Item) -> Render:
+    return RenderExtension.ext(ext_item).renders["thumbnail"]
+
+
+def test_collection_stac_extensions(ext_collection: pystac.Collection) -> None:
+    assert RenderExtension.has_extension(ext_collection)
+
+
+def test_item_stac_extensions(ext_item: pystac.Item) -> None:
+    assert RenderExtension.has_extension(ext_item)
+
+
+def test_ext_raises_if_item_does_not_conform(item: pystac.Item) -> None:
+    with pytest.raises(pystac.errors.ExtensionNotImplemented):
+        RenderExtension.ext(item)
+
+
+def test_ext_raises_if_collection_does_not_conform(
+    collection: pystac.Collection,
+) -> None:
+    with pytest.raises(pystac.errors.ExtensionNotImplemented):
+        RenderExtension.ext(collection)
+
+
+def test_ext_raises_on_catalog(catalog: pystac.Catalog) -> None:
+    with pytest.raises(
+        pystac.errors.ExtensionTypeError,
+        match="RenderExtension does not apply to type 'Catalog'",
+    ):
+        RenderExtension.ext(catalog)  # type: ignore
+
+
+def test_item_to_from_dict(ext_item_uri: str, ext_item: pystac.Item) -> None:
+    with open(ext_item_uri) as f:
+        d = json.load(f)
+    actual = ext_item.to_dict(include_self_link=False)
+    assert actual == d
+
+
+def test_collection_to_from_dict(
+    ext_collection_uri: str, ext_collection: pystac.Collection
+) -> None:
+    with open(ext_collection_uri) as f:
+        d = json.load(f)
+    actual = ext_collection.to_dict(include_self_link=False)
+    assert actual == d
+
+
+def test_add_to_item(item: pystac.Item) -> None:
+    assert not RenderExtension.has_extension(item)
+    RenderExtension.add_to(item)
+
+    assert RenderExtension.has_extension(item)
+
+
+def test_add_to_collection(collection: pystac.Collection) -> None:
+    assert not RenderExtension.has_extension(collection)
+    RenderExtension.add_to(collection)
+
+    assert RenderExtension.has_extension(collection)
+
+
+def test_get_render_values(thumbnail_render: Render) -> None:
+    assert thumbnail_render.title == "Thumbnail"
+    assert thumbnail_render.assets == ["B04", "B03", "B02"]
+    assert thumbnail_render.rescale == [[0, 150]]
+    assert thumbnail_render.colormap_name == "rainbow"
+    assert thumbnail_render.resampling == "bilinear"
+    assert thumbnail_render.properties.get("bidx") == [1]
+    assert thumbnail_render.properties.get("width") == 1024
+    assert thumbnail_render.properties.get("height") == 1024
+    assert thumbnail_render.properties.get("bands") == ["B4", "B3", "B2"]
```

Comment:
The extension spec describes [the Render object as "open ended"](https://github.com/stac-extensions/render#:~:text=The%20render%20object%20is%20open%20ended%2C%20so%20additional%20fields%20can%20be%20provided%20according%20to%20the%20needs%20of%20the%20rendering%20application). Is this the proper way to access additional attributes not defined as required by the spec?

Reply:
Yup, seems fine to me!